src/coreclr/vm/threadsuspend.cpp

```diff
@@ -6062,9 +6062,8 @@ void ThreadSuspend::Initialize()
 #ifdef FEATURE_HIJACK
 #if defined(TARGET_UNIX)
     ::PAL_SetActivationFunction(HandleSuspensionForInterruptedThread, CheckActivationSafePoint);
-#elif defined(TARGET_WINDOWS) && defined(TARGET_AMD64)
-    // Only versions of Windows that have the special user mode APC have a correct implementation of the return address hijack handling
-    if (Thread::UseSpecialUserModeApc())
+#elif defined(TARGET_WINDOWS)
+    if (Thread::AreCetShadowStacksEnabled())
```

Comment:
If you change this condition like this, the conditions in `STATUS_RETURN_ADDRESS_HIJACK_ATTEMPT` handling can be simplified to `if (Thread::AreCetShadowStacksEnabled() && (pExceptionInfo->ExceptionRecord->ExceptionCode == STATUS_RETURN_ADDRESS_HIJACK_ATTEMPT))`

Reply:
I thought about this and decided not to tie one condition to another. 
`STATUS_RETURN_ADDRESS_HIJACK_ATTEMPT` path can work if CET is not enabled. We had it that way, although it was a bit wasteful - lots of exceptions thrown/caught on every GC. 

I can see such scenario enabled though, at least for test purposes. I did that a few times by turning the condition here into `if(true)`, so I was hesitant to condition the ability to handle `STATUS_RETURN_ADDRESS_HIJACK_ATTEMPT` on enablement of shadow stacks.  

Perf savings from not looking at ExceptionCode  are probably very minor.

Reply:
if we condition ability to handle `STATUS_RETURN_ADDRESS_HIJACK_ATTEMPT` on enablement of CET, then some code inside support for `STATUS_RETURN_ADDRESS_HIJACK_ATTEMPT` can also be simplified. - the places under  
```c
if (areCetShadowStacksEnabled)
```

Reply:
Ok, the handling of `STATUS_RETURN_ADDRESS_HIJACK_ATTEMPT ` can stay as is then.

I have noticed that there is a bug in `STATUS_RETURN_ADDRESS_HIJACK_ATTEMPT ` handling that will show up when multiple runtimes in the process use it. We should bail if the thread is null or if we are not in cooperative mode. We know that it is some other code triggering `STATUS_RETURN_ADDRESS_HIJACK_ATTEMPT` in that case.

Reply:
> We should bail if the thread is null or if we are not in cooperative mode.

I think that would still allow for scenario where we try handling somebody's else hijacked method. 
We may need to check for the code ownership - like have `IsManaged()` somewhere.

Reply:
Although we would not be cooperative if running somebody else's code.

Reply:
> Although we would not be cooperative if running somebody else's code.

Right